daemon/containerd/image_list.go

```diff
@@ -77,29 +78,48 @@ func (i *ImageService) Images(ctx context.Context, opts types.ImageListOptions)
 			continue
 		}
 
-		platforms, err := images.Platforms(ctx, contentStore, img.Target)
-		if err != nil {
-			return nil, err
-		}
-
-		for _, platform := range platforms {
-			image, chainIDs, err := i.singlePlatformImage(ctx, contentStore, img, platform)
-			if err != nil {
-				return nil, err
-			}
-			if image == nil {
-				continue
+		err := images.Walk(ctx, images.HandlerFunc(func(ctx context.Context, desc v1.Descriptor) ([]v1.Descriptor, error) {
+			if images.IsIndexType(desc.MediaType) {
+				return images.Children(ctx, contentStore, desc)
 			}
 
-			summaries = append(summaries, image)
+			if images.IsManifestType(desc.MediaType) {
+				// Ignore attestation manifests
+				if dockerType, ok := desc.Annotations[attestation.DockerAnnotationReferenceType]; ok {
+					if dockerType == attestation.DockerAnnotationReferenceTypeDefault {
+						return nil, nil
+					}
+				}
```

Comment:
```suggestion
				if desc.Annotations[attestation.DockerAnnotationReferenceType] == attestation.DockerAnnotationReferenceTypeDefault {
					return nil, nil
				}
```

Or maybe even

```suggestion
				if _, ok := desc.Annotations[attestation.DockerAnnotationReferenceType]; ok {
					return nil, nil
				}
```

Reply:
Nice gold. We probably should check if `desc.Annotations` is not nil though? It wasn't nil when in my tests, but not sure if it's guaranteed.

Reply:
🏌️ It works with nil https://go.dev/play/p/bg_Nii2Xkv7

Reply:
🤯 TIL, thanks!
I got bitten too many times by this one: https://go.dev/play/p/kkNo7hg0r_H